components/SpinButton.vue

```diff
@@ -91,6 +100,33 @@ const props = defineProps({
     required: false,
     default: null,
   },
+  showSpinner: {
```

Comment:
I'm not convinced that we need `showSpinner`.  Generally:

- The event handler for `handle` can be async.
- The button should only spin while that handler is executing.  
- We should deal with that within this component.
- That means the parent doesn't need a separate local which it sets and resets inside the handler and then passes into this component to make the spinner show.
- This makes the parent code cleaner.  

It's possible that there may be some places where that approach doesn't work, but I think there are some in this PR where it does - `UserSearch` for example.  

So please review the changes and try to remove the local variables in the parents, and see what's left.

Reply:
You can have async event handlers, sure. What you can't is to have async emits. Emit is a coid function and you can't await the handler which listens for that emit. There are different patterns around that, which we can discuss in a separate issue I've created for refactoring.

Reply:
Ah, of course, you're right.

Ok, ignore that feedback - I'll carry on with reviewing.